aptos-move/framework/supra-framework/sources/pbo_delegation_pool.move

```diff
@@ -1601,10 +1586,237 @@ module supra_framework::pbo_delegation_pool {
+            );
+        };
+
+        // Synchronize the delegation and stake pools before any user operation.
+        synchronize_delegation_pool(pool_address);
+
+        // Ensure that each `delegator` has enough active stake to cover the corresponding `stake_to_lock`.
+        let new_amounts_to_lock = vector::zip_map(
+            delegators,
+            max_stakes_to_lock,
+            |delegator, max_stake_to_lock| {
+                let (active, inactive, pending_inactive) = get_stake(pool_address, delegator);
+                let already_locked = 0;
+                let pool: &mut DelegationPool = borrow_global_mut<DelegationPool>(pool_address);
+
+                // See if the delegator already has a locked allocation.
+                if (table::contains(&pool.principle_stake, delegator)) {
+                    already_locked = *table::borrow(&pool.principle_stake, delegator);
+                };
+
+                if (max_stake_to_lock > already_locked) {
+                    let max_new_stake_to_lock = max_stake_to_lock - already_locked;
+                    // All `already_locked` stake must be `active`.
+                    let lockable_active_stake = active - already_locked;
+
+                    // Ensure that all stake to be locked is made `active`.
+                    if (lockable_active_stake < max_new_stake_to_lock) {
+                        // The amount to lock can be covered by reactivating some previously unlocked stake.
+                        // Only reactivate the required amount to avoid unnecessarily interfering with 
+                        // in-progress withdrawals.
+                        let amount_to_reactivate = max_new_stake_to_lock - lockable_active_stake;
+
+                        // Ensure that we do not try to reactivate more than the available `pending_inactive` stake.
+                        // This should be enforced by functions within `authorized_reactivate_stake`, but checking
+                        // again here makes the correctness of this function easier to reason about.
+                        if (amount_to_reactivate > pending_inactive) {
+                            amount_to_reactivate = pending_inactive;
+                        };
+
+                        // Reactivate the required amount of `pending_inactive` stake first, if there is any.
+                        authorized_reactivate_stake(delegator, pool_address, amount_to_reactivate);
+                        let active_and_pending_inactive = lockable_active_stake + pending_inactive;
+                        
+                        if (active_and_pending_inactive < max_new_stake_to_lock) {
+                            // Need to reactivate some of the `inactive` stake.
+                            let amount_to_withdraw = max_new_stake_to_lock - active_and_pending_inactive;
+                            // Withdraw the minimum required amount to the admin's address.
+                            admin_withdraw(
```

Comment:
there should be a check that `inactive >= amount_to_withdraw` otherwise this may fail.